nntrainer/compiler/tflite_interpreter.cpp

```diff
@@ -202,33 +204,49 @@ using TfOpNodes = std::vector<TfOpNode>;
 /**
  * @brief Bidirectional Index map
  *
- * @tparam T type of a underlying value, please note that T will be copied, so
- * please use this for pointers and primitive values that is okay to copy
+ * @tparam Key type of a underlying hashable value, please note that T will be
+ * copied, so please use this for pointers and primitive values that is okay to
+ * copy
+ * @tparam Data data type to be stored inside the vector, if not given, same as
+ * KeyType
  */
-template <typename T> class BidirectionalIndexMap {
+template <typename KeyType, typename DataType = KeyType>
+class BidirectionalIndexMap {
```

Comment:
Changed IndexMap to embrace KeyType and DataType if they are different, else the interface remains same.